src/Map/assets/dist/abstract_map_controller.js

```diff
@@ -0,0 +1,43 @@
+import { Controller } from '@hotwired/stimulus';
+
+class default_1 extends Controller {
+    constructor() {
+        super(...arguments);
+        this.markers = [];
+        this.infoWindows = [];
+    }
+    initialize() { }
+    connect() {
+        this.dispatchEvent('pre-connect', {});
+        const { center, zoom, options, markers, fitBoundsToMarkers } = this.viewValue;
+        this.map = this.doCreateMap({ center, zoom, options });
+        markers.forEach((marker) => this.createMarker(marker));
+        if (fitBoundsToMarkers) {
+            this.doFitBoundsToMarkers();
+        }
+        this.dispatchEvent('connected');
+    }
+    createMarker(definition) {
+        this.dispatchEvent('marker:before-create', { definition });
+        const marker = this.doCreateMarker(definition);
+        this.dispatchEvent('marker:after-create', { marker });
+        this.markers.push(marker);
+        return marker;
+    }
+    createInfoWindow({ definition, marker, }) {
+        this.dispatchEvent('info-window:before-create', { definition, marker });
+        const infoWindow = this.doCreateInfoWindow({ definition, marker });
+        this.dispatchEvent('info-window:after-create', { infoWindow, marker });
+        this.infoWindows.push(infoWindow);
+        return infoWindow;
+    }
```

Comment:
I think you now can remove everything exposed about info-window (marker events are enough and allow access to infowindow)

Reply:
The method is supposed to be protected, because I wanted it to be called by children implementation and without duplicating code.

But yes, TypeScript generates the code as a public function and so is not protected from being called from outside.